python/lsst/meas/base/flagHandler.cc

```diff
@@ -32,36 +34,90 @@ namespace lsst {
 namespace meas {
 namespace base {
 
-PYBIND11_PLUGIN(flagHandler) {
-    py::module mod("flagHandler");
+namespace {
+
+void declareFlagDefinition(py::module &mod) {
+    py::class_<FlagDefinition, std::shared_ptr<FlagDefinition>> cls(mod, "FlagDefinition");
 
-    /* Module level */
-    py::class_<FlagDefinition> clsFlagDefinition(mod, "FlagDefinition");
-    py::class_<FlagHandler> clsFlagHandler(mod, "FlagHandler");
+    cls.def(py::init<>());
+    cls.def(py::init<std::string, std::string, std::size_t>(), "name"_a, "doc"_a,
+            "number"_a = FlagDefinition::number_undefined);
 
-    /* Member types and enums */
-    // Anonymous enum cannot be represented directly
-    clsFlagHandler.attr("FAILURE") = py::cast(static_cast<int>(FlagHandler::FAILURE));
+    cls.def_readwrite("name", &FlagDefinition::name);
+    cls.def_readwrite("doc", &FlagDefinition::doc);
+    cls.def_readwrite("number", &FlagDefinition::number);
+
+    cls.def("__eq__", &FlagDefinition::operator==, py::is_operator());
+    cls.def("__ne__", &FlagDefinition::operator!=, py::is_operator());
+}
 
-    /* Constructors */
-    clsFlagDefinition.def(py::init<>());
-    clsFlagDefinition.def(py::init<std::string, std::string>(), "_name"_a, "_doc"_a);
+void declareFlagDefinitionList(py::module &mod) {
+    py::class_<FlagDefinitionList, std::shared_ptr<FlagDefinitionList>> cls(mod, "FlagDefinitionList");
+
+    cls.def(py::init<>());
+    cls.def(py::init<std::initializer_list<FlagDefinition> const &>());
+
+    cls.def("__getitem__",
+            [](FlagDefinitionList const &self, int i) {
+                auto cind = lsst::utils::python::cppIndex(self.size(), i);
+                return self[cind];
+            },
+            py::is_operator());
```

Comment:
Are you sure? I thought it would result in `__getitem__` giving the expected behavior if the type of the index-like argument was wrong (e.g. a string provided when an int was expected).